src/main/java/roomescape/admin/service/ReservationService.java

```diff
@@ -4,33 +4,74 @@
 import roomescape.admin.dto.ReadReservationResponse;
 import roomescape.admin.dto.SaveReservationRequest;
 import roomescape.admin.entity.Reservation;
+import roomescape.admin.entity.ReservationTime;
 import roomescape.admin.repository.ReservationRepository;
+import roomescape.admin.repository.ReservationTimeRepository;
+import roomescape.common.exception.CustomException;
 
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.List;
 
 @Service
 public class ReservationService {
 
     private final ReservationRepository reservationRepository;
+    private final ReservationTimeRepository reservationTimeRepository;
 
-    public ReservationService(ReservationRepository reservationRepository){
+    public ReservationService(ReservationRepository reservationRepository,
+                              ReservationTimeRepository reservationTimeRepository){
         this.reservationRepository = reservationRepository;
+        this.reservationTimeRepository = reservationTimeRepository;
     }
 
     public List<ReadReservationResponse> readReservation() {
-        List<Reservation> reservation = this.reservationRepository.readReservation();
+        List<Reservation> reservation = this.reservationRepository.findAll();
 
         return ReadReservationResponse.from(reservation);
     }
 
     public ReadReservationResponse saveReservation(SaveReservationRequest saveReservationRequest) {
-        Long id = this.reservationRepository.saveReservation(saveReservationRequest);
-        Reservation reservation = this.reservationRepository.readReservationById(id);
+        if(isDuplicatedReservation(saveReservationRequest)){
+            throw new CustomException("중복된 예약이 존재합니다.");
+        }
+
+        if(isUnderDate(saveReservationRequest)){
+            throw new CustomException("예약 시간은 현재 시간보다 이전일 수 없습니다.");
+        }
+
+        Long id = this.reservationRepository.save(saveReservationRequest);
+        Reservation reservation = this.reservationRepository.findById(id);
 
         return ReadReservationResponse.from(reservation);
     }
 
+    private boolean isUnderDate(SaveReservationRequest saveReservationRequest) {
+        ReservationTime reservationTime = this.reservationTimeRepository.findById(saveReservationRequest.timeId());
+        String startAt = reservationTime.getStartAt();
+        LocalDateTime reserve = LocalDateTime.parse(saveReservationRequest.date() + " " + startAt, DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
+
+        if (reserve.isBefore(LocalDateTime.now())) {
+            return true;
+        }
+
+        return false;
```

Comment:
바로 return 해줄 수 있을 것 같아용

```suggestion
        return reserve.isBefore(LocalDateTime.now());
```